src/asset-swapper/utils/market_operation_utils/aave_reserves_cache.ts

```diff
@@ -36,30 +57,52 @@ export interface AaveReserve {
     };
 }
 
+export interface AaveV3Reserve {
+    id: string;
+    underlyingAsset: string;
+    aToken: {
+        id: string;
+    };
+    pool: {
+        id: string;
+        pool: string;
+    };
+}
+
 interface Cache {
-    [key: string]: AaveReserve[];
+    [key: string]: (AaveV2Reserve | AaveV3Reserve)[];
 }
 
 const RESERVES_REFRESH_INTERVAL_MS = 30 * constants.ONE_MINUTE_MS;
-
 /**
- * Fetches Aave V2 reserve information from the official subgraph(s).
+ * Fetches Aave V2/V3 reserve information from the official subgraph(s).
  * The reserve information is updated every 30 minutes and cached
  * so that it can be accessed with the underlying token's address
  */
-export class AaveV2ReservesCache {
+export class AaveReservesCache {
```

Comment:
could we use generics here? AaveReservesCache<T>? would make return type of get simpler. or could we unify the different types of reserves they are both pretty similar. The return type of (AaveV2Reserve | AaveV3Reserve)[] makes handling the data a little awkward downstream.

Reply:
Yeah, I agree that the return type is a little bit awkward downstream. I attempted to use generics here where the generic type would be either AaveV2Reserve or AaveV3Reserve but I couldn't figure out how to change the query depending on whether the generic type is AaveV2Reserve or AaveV3Reserve. Open to any suggestions on how to make this work with generics.